test/src/telemetry/test_telemetry.c

```diff
@@ -76,17 +82,33 @@ test_factory(ConnectionType type, int status, char *host, int port)
 		ts_connection_mock_set_recv_buf(conn, test_string, strlen(test_string));
 #endif
 
-	req = build_request(status);
+	int retries = 0;
+	while (retries < INVALID_STATUS_RETRIES)
+	{
+		req = build_request(status);
 
-	rsp = ts_http_response_state_create();
+		rsp = ts_http_response_state_create();
 
-	err = ts_http_send_and_recv(conn, req, rsp);
+		err = ts_http_send_and_recv(conn, req, rsp);
 
-	ts_http_request_destroy(req);
-	ts_connection_destroy(conn);
+		ts_http_request_destroy(req);
 
-	if (err != HTTP_ERROR_NONE)
-		elog(ERROR, "%s", ts_http_strerror(err));
+		if (err != HTTP_ERROR_NONE)
+			elog(ERROR, "%s", ts_http_strerror(err));
```

Comment:
If some intermittent http error happen it will not raise an error and the test continue be flaky??

Reply:
This change is just for wrong status code like Bad Gateway (502) which happens from time to time. What you are referring to is a request error which I haven't really seen.